File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
from collections import deque

input = sys.stdin.readline

N, M, V = map(int, input().split())

dfs_result = []
bfs_result = []


def dfs(graph, v, visited):
    visited[v] = True
    dfs_result.append(v)
    for i in graph[v]:
        if not visited[i]:
            visited[i] = True
            dfs(graph, i, visited)


def bfs(graph, v, visited):
    visited[v] = True
    queue = deque([v])
    while queue:
        q = queue.popleft()
        bfs_result.append(q)
        for i in graph[q]:
            if not visited[i]:
                queue.append(i)
                visited[i] = True


graph = [[] for _ in range(N + 1)]

for _ in range(M):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

for i in graph:
    i.sort()

visited_dfs = [False] * (N + 1)
visited_bfs = [False] * (N + 1)

dfs(graph, V, visited_dfs)
bfs(graph, V, visited_bfs)
print(*dfs_result)
print(*bfs_result)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let inputLines = [];
let currentLine = 0;

rl.on('line', (line) => {
    inputLines.push(line.trim());
});

rl.on('close', () => {
    const [N, M, V] = inputLines[0].split(' ').map(Number);
    
    let dfsResult = [];
    let bfsResult = [];
    
    function dfs(graph, v, visited) {
        visited[v] = true;
        dfsResult.push(v);
        for (let i of graph[v]) {
            if (!visited[i]) {
                visited[i] = true;
                dfs(graph, i, visited);
            }
        }
    }
    
    function bfs(graph, v, visited) {
        visited[v] = true;
        let queue = [v];
        
        while (queue.length > 0) {
            let q = queue.shift();
            bfsResult.push(q);
            for (let i of graph[q]) {
                if (!visited[i]) {
                    queue.push(i);
                    visited[i] = true;
                }
            }
        }
    }
    
    let graph = Array.from({ length: N + 1 }, () => []);
    
    for (let i = 1; i <= M; i++) {
        const [a, b] = inputLines[i].split(' ').map(Number);
        graph[a].push(b);
        graph[b].push(a);
    }
    
    for (let i = 0; i <= N; i++) {
        graph[i].sort((a, b) => a - b);
    }
    
    let visitedDfs = new Array(N + 1).fill(false);
    let visitedBfs = new Array(N + 1).fill(false);
    
    dfs(graph, V, visitedDfs);
    bfs(graph, V, visitedBfs);
    console.log(dfsResult.join(' '));
    console.log(bfsResult.join(' '));
});